### PR TITLE
App feature usage info in UserAgent

### PR DESCRIPTION
**Pull request by ivanbogdanov**

### CODE REVIEW

**Comment by huminzhi** [comment]
Can one of the admins verify this patch?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.h [hunk 1]
```diff
@@ -41,6 +41,13 @@ typedef NS_ENUM(NSUInteger, SFAppType) {
     kSFAppTypeReactNative
 };
 
+typedef NS_ENUM(NSUInteger, SFAppFeature) {
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.h]: I don't think SalesforceSDKCore should know about the other libraries like SmartStore etc.
So instead appFeature could be a string.

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.h]: I agree.

Comment (ivanbogdanov) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.h]: Fixed

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m [hunk 2]
```diff
@@ -79,6 +84,7 @@ + (instancetype)sharedManager
     static SalesforceSDKManager *sdkManager = nil;
     dispatch_once(&pred , ^{
         uid = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+        features = [NSMutableSet setWithObjects:kSFAppFeatureOAuth, nil];
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m]: Should you set the oauth feature from the oauth code - I guess it's possible to do anonymous (unauthenticated) calls to the server.

Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m]: Agreed.  The OAuth setting should follow the same pattern that SmartStore does, as a consumer.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m [hunk 3]
```diff
@@ -41,6 +41,11 @@
 // Device id
 static NSString* uid = nil;
 
+static NSMutableSet *features;
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m]: This variable is only ever used in the singleton context.  It should be represented as a property on the singleton.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m [hunk 4]
```diff
@@ -41,6 +41,11 @@
...
+
+NSString * const kSFAppFeatureOAuth = @"UA";
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SalesforceSDKManager.m]: This can be a `static` constant, if it's not going to be external.

**Comment by ivanbogdanov** [comment]
Fixed. `UA` is reported on auth event

**Comment by ivanbogdanov** [comment]
Fixed.

**Comment by sgoldberg-sfdc** [comment]
Is there somewhere this concept of app feature is explained or documented?

**Comment by ivanbogdanov** [comment]
Done: Removed extra empty line from `SFOAuthCoordinator.m`

**Comment by ivanbogdanov** [comment]
@wmathurin @bhariharan

**Comment by wmathurin** [comment]
👍

**Pull request merged**